src/app/use-cases/counter.ts

```diff
@@ -43,11 +43,19 @@ export const selectors = {
 };
 
 export const thunks = {
-  incrementAsync: createAsyncThunk<number, number, AsyncThunkConfig>('counter/fetchCount', async (amount: number, { extra }) => {
-    const { data } = await extra.counterApi.fetchCount(amount);
+  ...(() => {
 
-    return data;
-  }),
+    const thunkName = "incrementAsync";
+
+    return {
+      [thunkName]: createAsyncThunk<number, number, AsyncThunkConfig>(`${name}/${thunkName}`, async (amount: number, { extra }) => {
+        const { data } = await extra.counterApi.fetchCount(amount);
+
+        return data;
+      })
+    };
+
+  })(),
```

Comment:
...This is why I no longer use `createAsyncThunk`. I don't think there is another way to write it that do not requires to keep tokens in sync.

Reply:
There isn't a problem when the action name ('counter/fetchCount') is not the same as the function name ('incrementAsync'), to be continued.

Reply:
> There isn't a problem when the action name ('counter/fetchCount')

You are right. I still feel that they should be matching though...